docs/contributing.md

```diff
@@ -1,18 +1,161 @@
-Contributing
-============
+# Contributing
 
-Testing
--------
+## Getting Started
+
+You’ll need a Kubernetes cluster to run against. You can use
+[KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run
+against a remote cluster.  **Note:** Your controller will automatically use the
+current context in your kubeconfig file (i.e. whatever cluster `kubectl
+cluster-info` shows).
+
+### Running on the cluster
+1. Install Instances of Custom Resources:
+
+```sh
+kubectl apply -f config/samples/
+```
+
+2. Build and push your image to the location specified by `IMG`:
+
+```sh
+make docker-build docker-push IMG=<some-registry>/dataplane-operator:tag
+```
+
+3. Deploy the controller to the cluster with the image specified by `IMG`:
+
+```sh
+make deploy IMG=<some-registry>/dataplane-operator:tag
+```
+
+### Uninstall CRDs
+To delete the CRDs from the cluster:
+
+```sh
+make uninstall
+```
+
+### Undeploy controller
+UnDeploy the controller to the cluster:
+
+```sh
+make undeploy
+```
+
+### How it works
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources untile the desired state is reached on the cluster
```

Comment:
- "provide" (singular) not "provides"
- until (not untile)

Reply:
thanks, fixed these.